File: src/main/java/com/jetbrains/gymusserverjava/memberships/dtos/requests/UpdateMemberRequestDto.java
package com.jetbrains.gymusserverjava.memberships.dtos.requests;

import jakarta.validation.constraints.*;
import org.springframework.web.multipart.MultipartFile;

public record UpdateMemberRequestDto(
        @NotBlank
        @NotNull
        @NotEmpty
        @Email
        String email,
        @NotBlank
        @NotNull
        @NotEmpty
        @Size(max = 25)
        String phoneNumber,
        @NotNull
        @NotEmpty
        @NotBlank
        String address,
        MultipartFile medicalCertificate,
        MultipartFile birthCertificate,
        MultipartFile personalPhoto,
        MultipartFile parentalAuthorization,
        @NotNull
        @NotEmpty
        @NotBlank
        String membershipType
) {
}

File: src/main/java/com/jetbrains/gymusserverjava/memberships/dtos/requests/UpdateMemberRequestDto.kt
package com.jetbrains.gymusserverjava.memberships.dtos.requests

import jakarta.validation.constraints.*
import org.springframework.web.multipart.MultipartFile

@JvmRecord
data class UpdateMemberRequestDto(
    @NotBlank
    @NotNull
    @NotEmpty
    @Email
    @JvmField val email: String,

    @NotBlank
    @NotNull
    @NotEmpty
    @Size(max = 25)
    @JvmField val phoneNumber: String,

    @NotNull
    @NotEmpty
    @NotBlank
    @JvmField val address: String,
    @JvmField val medicalCertificate: MultipartFile,
    @JvmField val birthCertificate: MultipartFile,
    @JvmField val personalPhoto: MultipartFile,
    @JvmField val parentalAuthorization: MultipartFile?,

    @NotNull
    @NotEmpty
    @NotBlank
    @JvmField val membershipType: String
)
